### DISCUSSION TITLE
Server route returns 404 error

### DISCUSSION THREAD

**Discussion by melokki** [nuxt/nuxt#16218] [Questions] [answered]
Hello everyone

I am building a ssr website and I am facing a issue I cannot understand

I've created some handles inside `server` folder for my routes, but every time a route is used with `useFetch` or any other data fetching composable I get in console the following error:

```
{
  "url": "/api/users/",
  "statusCode": 404,
  "statusMessage": "Cannot find any route matching /api/users/",
  "message": "Cannot find any route matching /api/users/",
  "stack": ""
}
```

It's even more confusing as the actual api route is hit, I now that because I can save/get data into/from my database, but I am getting these errors in console for every api route

Is it something I am doing wrong?

**Comment by TheAlexLichter** [comment, marked as answer]
**Update**: This is not the case anymore since h3 v1.7.0 - https://github.com/unjs/h3/pull/409

Old: If you do not return anything in your route, a 404 is shown. See https://github.com/nuxt/framework/issues/8554

**Reply by melokki** [reply]
thank you very much for your answer. I managed to fix the issue :pray:

**Reply by miumerfarooq** [reply]
Thank you brother for helping ;)
